peerconnection.go

```diff
@@ -557,67 +574,89 @@ func (pc *PeerConnection) CreateOffer(options *OfferOptions) (SessionDescription
 		}
 	}
 
-	isPlanB := pc.configuration.SDPSemantics == SDPSemanticsPlanB
-	if pc.currentRemoteDescription != nil {
-		isPlanB = descriptionIsPlanB(pc.RemoteDescription())
-	}
+	var (
+		d     *sdp.SessionDescription
+		offer SessionDescription
+		err   error
+	)
 
-	// include unmatched local transceivers
-	if !isPlanB {
-		// update the greater mid if the remote description provides a greater one
+	// This may be necessary to recompute if, for example, createOffer was called when only an
+	// audio RTCRtpTransceiver was added to connection, but while performing the in-parallel
+	// steps to create an offer, a video RTCRtpTransceiver was added, requiring additional
+	// inspection of video system resources.
+	for {
+		// We cache current transceivers to ensure they aren't
+		// mutated during offer generation. We later check if they have
+		// been mutated and recompute the offer if necessary.
+		currentTransceivers := pc.GetTransceivers()
+
+		// in-parallel steps to create an offer
+		// https://w3c.github.io/webrtc-pc/#dfn-in-parallel-steps-to-create-an-offer
+		isPlanB := pc.configuration.SDPSemantics == SDPSemanticsPlanB
```

Comment:
the only modification between here and line `653` is to use `currentTransceivers` rather than `pc.GetTransceivers()`